### PR TITLE
refactor(spec): add uint8 for int encoding and templatize

### PR DESCRIPTION
**Pull request by archerzz**
- .net doesn't support uint32 yet, so use uint8
- use a value which will be negative in signed int

# Cadl Ranch Contribution Checklist:

- [ ] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [ ] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [ ] I have written a [mock API](../docs/writing-mock-apis.md)
- [ ] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 1]
```diff
@@ -2246,24 +2246,24 @@ Expected response body:
...
+- Endpoint: `post /encode/numeric/property/uint8`
 
-Test operation with request and response model contains property of uint32 type with string encode.
+Test operation with request and response model contains property of uint8 type with string encode.
 Expected request body:
 
 ```json
 {
-  "value": "1"
+  "value": "255"
```
Comment (iscai-msft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:2258]: why do we want to change it to `uint8`? Would it be better to add a test for `uint8` instead

Comment (archerzz) [reply packages/cadl-ranch-specs/cadl-ranch-summary.md:2279]: Thanks. I've added back `uint32` test case.

File: packages/cadl-ranch-specs/http/encode/numeric/main.tsp [hunk 2]
```diff
@@ -62,4 +62,29 @@ namespace Property {
...
+    ```
+    Expected response body:
+    ```json
+    {
+      "value": "255"
+    }
+    ```
+    """)
+  @post
+  op uint8AsStringOptional(@body body: Uint8AsStringProperty): Uint8AsStringProperty;
```
Comment (iscai-msft) [inline packages/cadl-ranch-specs/http/encode/numeric/main.tsp:84]: do you want to add one for `uint8AsString`? Not sure if it's needed, it's just that corresponds more directly with what the tests are for `uint32`. They can also be templatized this way, but you can do that in a separate pr

Comment (archerzz) [reply packages/cadl-ranch-specs/http/encode/numeric/main.tsp:84]: the reason I add `uint8` is because .net generator doesn't support uint32 yet, so I need an `uint8` case

Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/encode/numeric/main.tsp:84]: I think Issabella is only about the "optional" part.

I've added uint32 as optional property, because Java type is different on required/optional for primitive type. You may not need this "optional" for uint8

Comment (archerzz) [reply packages/cadl-ranch-specs/http/encode/numeric/main.tsp]: OK. I've templatize the test cases.

File: packages/cadl-ranch-specs/http/encode/numeric/main.tsp [hunk 3]
```diff
@@ -13,53 +13,60 @@ namespace Encode.Numeric;
...
+
+  alias SendUint8AsString = SendIntAsString<uint8, "255", Uint8AsStringProperty>;
+
+  @route("/uint8")
+  op uint8AsStringOptional is SendUint8AsString.sendIntAsString;
+
+  model Uint8AsStringProperty {
+    @encode(string)
+    value: uint8;
+  }
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/encode/numeric/main.tsp:44]: This value seems not "optional"

Comment (archerzz) [reply packages/cadl-ranch-specs/http/encode/numeric/main.tsp]: ah, removed the `Optional` suffix.

File: packages/cadl-ranch-specs/http/encode/numeric/main.tsp [hunk 4]
```diff
@@ -13,53 +13,60 @@ namespace Encode.Numeric;
...
-    Expected response body:
-    ```json
-    {
-      "value": "1"
-    }
-    ```
-    """)
-  @post
-  op uint32AsStringOptional(@body body: Uint32AsStringProperty): Uint32AsStringProperty;
+  op uint32AsStringOptional is SendUint32AsString.sendIntAsString;
```
Comment (iscai-msft) [inline packages/cadl-ranch-specs/http/encode/numeric/main.tsp:29]: nit: we shouldn't need the alias, you should just be able to directly do `op uint32AsStringOptions is SendIntAsString<uint32, "1", Uint32AsStringProperty>`

**Review by iscai-msft — APPROVED** [review]

**Review by weidongxu-microsoft — APPROVED** [review]

File: packages/cadl-ranch-specs/http/encode/numeric/main.tsp (hunk 4 above)
Comment (archerzz) [reply packages/cadl-ranch-specs/http/encode/numeric/main.tsp:29]: Looks like the compiler won't allow it.
image

I'll merge this first. We can create a separate PR to refactor if needed.
